npm/src/lib/definitions.js

```diff
@@ -5,67 +5,97 @@ import {fs, https, path} from "./node.js";
-  const dirItemStats = await P.all(
-    dirItems.map(async (item) => fs.stat(path.join(pkgDir, item)))
-  );
-
-  return dirItems.filter((item, idx) => dirItemStats[idx].isDirectory());
-};
+export const VERSIONED_NAME_RE = /^(.*)-([><]?=)?v([0-9]*|x)\.([0-9]*|x)\.([0-9]*|x)$/;
+function dirNameToVersionedName(dirPath): VersionedName {
+  const dirName = path.basename(dirPath);
+  const dirNameParts = dirName.match(VERSIONED_NAME_RE);
+  if (dirNameParts == null) {
+    throw new Error(
+      "'" + dirName + "' has a malformed name. Expected a name formatted as " +
+      "`<PKGNAME>-v<MAJOR>.<MINOR>.<PATCH>`."
+    );
+  }
+  let [_, name, range, major, minor, patch] = dirNameParts;
 
-export async function getFlowVersionsForPkgVersion(
-  pkgName: string,
-  pkgVer: string
-): Promise<Array<string>> {
-  const pkgDir = await getPkgDir(pkgName);
-  const pkgVerDir = path.join(pkgDir, pkgVer);
-  if (!(await fs.exists(pkgVerDir))) {
+  if (range != null && range !== ">=" && range !== "<=") {
     throw new Error(
-      "'" + pkgName + "@" + pkgVer + "' does not exist in the " +
-      "definitions library!"
+      "'" + dirName + "': Invalid package version range: " + range
     );
   }
 
-  const dirItems = await fs.readdir(pkgVerDir);
-  const dirItemStats = await P.all(
-    dirItems.map(async (item) => fs.stat(path.join(pkgVerDir, item)))
-  );
+  major = _validateVersionPartStr(major, dirName, "major");
+  minor = _validateVersionPartStr(minor, dirName, "minor");
+  patch = _validateVersionPartStr(patch, dirName, "patch");
+
+  return {
+    name,
+    version: {range, major, minor, patch},
+    path: dirPath,
+  };
+};
 
+export async function getFlowVersionsForPackage(
+  pkgVer: VersionedName
+): Promise<Array<VersionedName>> {
+  const dirItems = await fs.readdir(pkgVer.path);
+  const dirItemStats = await P.all(dirItems.map(name => {
+    return fs.stat(path.join(pkgVer.path, name));
+  }));
   return dirItems
     .filter((_, idx) => dirItemStats[idx].isDirectory())
-    .map((itemName) => itemName.substr('flow-'.length));
+    .map(dirName => {
+      const dirPath = path.join(pkgVer.path, dirName);
+      return dirName !== "flow-all" ? dirNameToVersionedName(dirPath) : {
+        name: "flow",
+        version: {major: "x", minor: "x", patch: "x"},
+        path: dirPath
+      };
+    });
+};
+
+export async function getPackages(): Promise<Array<VersionedName>> {
+  return (await fs.readdir(DEFINITIONS_DIR)).map(
+    dirName => dirNameToVersionedName(path.join(DEFINITIONS_DIR, dirName))
+  );
 };
 
-export function isTestFileName(
-  itemName: string,
-  itemStats: fs.Stats
-): boolean {
+export function versionToString(ver: Version): string {
+  const rangeStr = ver.range ? ver.range : '';
   return (
-    itemStats.isFile()
-    && itemName.substr(0, 'test-'.length) === 'test-'
-    && path.extname(itemName) === '.js'
+    rangeStr + 'v' + String(ver.major) + '.' + String(ver.minor) + '.' +
```

Comment:
Why not something like:
```
const { range, major, minor, patch } = ver;
const rangeStr = range ? range : '';

return `${rangeStr}v${major}.${minor}.${patch}`;
);
```

I mean, it's not really an issue, but since this tool relies a lot of path-building, I find this much easier to read...